packages/react-native-reanimated/src/createAnimatedComponent/NativeEventsManager.ts

```diff
@@ -0,0 +1,144 @@
+  private _componentOptions?: ComponentOptions;
+  private _eventViewTag = -1;
+
+  constructor(component: ManagedAnimatedComponent, options?: ComponentOptions) {
+    this._managedComponent = component;
+    this._componentOptions = options;
+    this._eventViewTag = this.getEventViewTag();
+  }
+
+  public attachEvents() {
+    this.executeForEachEventHandler(
+      this._managedComponent.props,
+      (key, handler) => {
+        handler.registerForEvents(this._eventViewTag, key);
+      }
+    );
+  }
+
+  public detachEvents() {
+    this.executeForEachEventHandler(
+      this._managedComponent.props,
+      (_key, handler) => {
+        handler.unregisterFromEvents(this._eventViewTag);
+      }
+    );
+  }
+
+  public updateEvents(
+    prevProps: AnimatedComponentProps<InitialComponentProps>
+  ) {
+    const computedEventTag = this.getEventViewTag();
+    // If the event view tag changes, we need to completely re-mount all events
+    if (this._eventViewTag !== computedEventTag) {
+      // Remove all bindings from previous props that ran on the old viewTag
+      this.executeForEachEventHandler(prevProps, (_key, handler) => {
+        handler.unregisterFromEvents(this._eventViewTag);
+      });
+      // We don't need to unregister from current (new) props, because their events weren't registered yet
+      // Replace the view tag
+      this._eventViewTag = computedEventTag;
+      // Attach the events with a new viewTag
+      this.attachEvents();
+      return;
+    }
+
+    this.executeForEachEventHandler(prevProps, (key, prevHandler) => {
+      const newProp = this._managedComponent.props[key];
+      if (!newProp) {
+        // Prop got deleted
+        prevHandler.unregisterFromEvents(this._eventViewTag);
+      } else if (
+        this.isWorkletEventHandler(newProp) &&
+        newProp.workletEventHandler !== prevHandler
+      ) {
+        // Prop got changed
+        prevHandler.unregisterFromEvents(this._eventViewTag);
+        newProp.workletEventHandler.registerForEvents(this._eventViewTag);
+      }
+    });
+
+    this.executeForEachEventHandler(
+      this._managedComponent.props,
+      (key, handler) => {
+        if (!prevProps[key]) {
+          // Prop got added
+          handler.registerForEvents(this._eventViewTag);
+        }
+      }
+    );
+  }
+
+  private isWorkletEventHandler(
+    prop: unknown
+  ): prop is WorkletEventHandlerHolder {
+    return (
+      has('workletEventHandler', prop) &&
+      prop.workletEventHandler instanceof WorkletEventHandler
+    );
+  }
+
+  private executeForEachEventHandler(
+    props: AnimatedComponentProps<InitialComponentProps>,
+    callback: (
+      key: string,
+      handler: InstanceType<typeof WorkletEventHandler>
+    ) => void
+  ) {
+    for (const key in props) {
+      const prop = props[key];
+      if (this.isWorkletEventHandler(prop)) {
+        callback(key, prop.workletEventHandler);
+      }
+    }
+  }
```

Comment:
This seems nice to avoid code duplication, but I find the manager itself hard to read with it. I'm considering if we should move to inline `Object.entries()` instead.